### PR TITLE
Matthew Moving Mini Project

### PR DESCRIPTION
**Pull request by MatthewGetachew**
Look at my pr

### CODE REVIEW

File: camp/mgetachew/src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  }
+
+  /**
+   * Use this method to define your button->command mappings. Buttons can be created by
+   * instantiating a {@link edu.wpi.first.wpilibj.GenericHID} or one of its subclasses ({@link
+   * edu.wpi.first.wpilibj.Joystick} or {@link XboxController}), and then passing it to a {@link
+   * edu.wpi.first.wpilibj2.command.button.JoystickButton}.
+   */
+  private void configureButtonBindings() {
+    JoystickButton motorForward = new JoystickButton(highLevelButtonBoard, 1);
```
Comment (josieammer) [inline camp/mgetachew/src/main/java/frc/robot/RobotContainer.java:45]: constant '1' should be in Constants.java

File: camp/mgetachew/src/main/java/frc/robot/RobotContainer.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    JoystickButton motorBackward = new JoystickButton(highLevelButtonBoard, 2);
```
Comment (josieammer) [inline camp/mgetachew/src/main/java/frc/robot/RobotContainer.java:46]: constant '2' should be in Constants.java

File: camp/mgetachew/src/main/java/frc/robot/commands/Backward.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands;
+
+import frc.robot.subsystems.MatthewsCoolMotor;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
+/** An example command that uses an example subsystem. */
```
Comment (josieammer) [inline camp/mgetachew/src/main/java/frc/robot/commands/Backward.java:10]: update this comment bloc.

File: camp/mgetachew/src/main/java/frc/robot/commands/Backward.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+public class Backward extends CommandBase {
+  @SuppressWarnings({"PMD.UnusedPrivateField", "PMD.SingularField"})
+  private final MatthewsCoolMotor m_subsystem;
+
+  /**
```
Comment (josieammer) [inline camp/mgetachew/src/main/java/frc/robot/commands/Backward.java:15]: update this comment bloc.

File: camp/mgetachew/src/main/java/frc/robot/commands/Forward.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands;
+
+import frc.robot.subsystems.MatthewsCoolMotor;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
+/** An example command that uses an example subsystem. */
```
Comment (josieammer) [inline camp/mgetachew/src/main/java/frc/robot/commands/Forward.java:10]: update this comment bloc, and the ones below.

File: camp/mgetachew/src/main/java/frc/robot/commands/Stop.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands;
+
+import frc.robot.subsystems.MatthewsCoolMotor;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
+/** An example command that uses an example subsystem. */
```
Comment (josieammer) [inline camp/mgetachew/src/main/java/frc/robot/commands/Stop.java:10]: update comment bloc and ones below.

File: camp/mgetachew/src/main/java/frc/robot/forwardCmd.java [hunk 7]
```diff
@@ -0,0 +1,5 @@
+package frc.robot;
```
Comment (josieammer) [inline camp/mgetachew/src/main/java/frc/robot/forwardCmd.java:1]: is this file needed?

File: camp/mgetachew/src/main/java/frc/robot/subsystems/MatthewsCoolMotor.java [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+  /** Creates a new RomiDrivetrain. */
+  public MatthewsCoolMotor() {
+  }
+
+  /**
+  * A method exposed to callers to store one ball into storage
+  */
+  public void footToTheFloorForward()
+  {
+    myMotor.set(1.0);
```
Comment (josieammer) [inline camp/mgetachew/src/main/java/frc/robot/subsystems/MatthewsCoolMotor.java:25]: you should pull this value out to Constants.java

File: camp/mgetachew/src/main/java/frc/robot/subsystems/MatthewsCoolMotor.java [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+public class MatthewsCoolMotor extends SubsystemBase {
+
+  private WPI_TalonSRX myMotor = new WPI_TalonSRX(Constants.myMotorCanId);
+
+  /** Creates a new RomiDrivetrain. */
+  public MatthewsCoolMotor() {
+  }
+
+  /**
+  * A method exposed to callers to store one ball into storage
```
Comment (josieammer) [inline camp/mgetachew/src/main/java/frc/robot/subsystems/MatthewsCoolMotor.java:21]: update this comment bloc. - applies to any public method, constructor and class

**Review by josieammer — CHANGES_REQUESTED** [review]

File: camp/mgetachew/src/main/java/frc/robot/commands/Backward.java [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+
+package frc.robot.commands;
+
+import frc.robot.subsystems.MatthewsCoolMotor;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
+/** An example command that uses an example subsystem. */
+public class Backward extends CommandBase {
+  @SuppressWarnings({"PMD.UnusedPrivateField", "PMD.SingularField"})
+  private final MatthewsCoolMotor m_subsystem;
```
Comment (mmcadamsms) [inline camp/mgetachew/src/main/java/frc/robot/commands/Backward.java:13]: I just read about final for the member here.  I would say that I don't usually see final used in this way.  Based on what I read:

_A blank final variable can be initialized inside an instance-initializer block or inside the constructor. If you have more than one constructor in your class then it must be initialized in all of them, otherwise, a compile-time error will be thrown._

Just an FYI.

File: camp/mgetachew/src/main/java/frc/robot/commands/Backward.java [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands;
+
+import frc.robot.subsystems.MatthewsCoolMotor;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
+/** An example command that uses an example subsystem. */
+public class Backward extends CommandBase {
+  @SuppressWarnings({"PMD.UnusedPrivateField", "PMD.SingularField"})
```
Comment (mmcadamsms) [inline camp/mgetachew/src/main/java/frc/robot/commands/Backward.java:12]: Not sure we want to get in the practice of suppressing warnings.  Lets bring them to the surface and then when we need them we put a comment in the code that describes why the warning needs to be suppressed.

File: camp/mgetachew/src/main/java/frc/robot/commands/Forward.java [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+public class Forward extends CommandBase {
+  @SuppressWarnings({"PMD.UnusedPrivateField", "PMD.SingularField"})
```
Comment (mmcadamsms) [inline camp/mgetachew/src/main/java/frc/robot/commands/Forward.java:12]: comment why suppression is necessary

File: camp/mgetachew/src/main/java/frc/robot/subsystems/MatthewsCoolMotor.java (hunk 9 above)
Comment (MatthewGetachew) [reply camp/mgetachew/src/main/java/frc/robot/subsystems/MatthewsCoolMotor.java]: Removed

File: camp/mgetachew/src/main/java/frc/robot/forwardCmd.java (hunk 7 above)
Comment (MatthewGetachew) [reply camp/mgetachew/src/main/java/frc/robot/forwardCmd.java]: its gone
